spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/PubSubBeanProcessor.java

```diff
@@ -0,0 +1,150 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.pubsub;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.cloud.spring.autoconfigure.core.GcpProperties;
+import com.google.cloud.spring.core.DefaultGcpProjectIdProvider;
+import com.google.cloud.spring.core.GcpProjectIdProvider;
+import com.google.cloud.spring.pubsub.core.PubSubConfiguration;
+import com.google.cloud.spring.pubsub.support.PubSubSubscriptionUtils;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+/**
+ * Class to register ThreadPoolTaskScheduler beans.
+ */
+public class PubSubBeanProcessor implements BeanDefinitionRegistryPostProcessor {
+
+	private ConfigurableEnvironment environment;
+
+	private ConcurrentHashMap<String, ThreadPoolTaskScheduler> threadPoolTaskSchedulerMap = new ConcurrentHashMap<>();
+
+	private ThreadPoolTaskScheduler globalScheduler;
+
+	public PubSubBeanProcessor(ConfigurableEnvironment environment) {
+		this.environment = environment;
+	}
+
+	@Override
+	public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry beanDefinitionRegistry) throws BeansException {
+		Binder binder = Binder.get(this.environment);
+		String cloudPropertiesPrefix = GcpProperties.class.getAnnotation(ConfigurationProperties.class)
+				.value();
+		GcpProperties gcpProperties = binder.bind(cloudPropertiesPrefix, GcpProperties.class)
+				.orElse(new GcpProperties());
+		GcpProjectIdProvider projectIdProvider = (gcpProperties.getProjectId() != null)
+				? gcpProperties::getProjectId
+				: new DefaultGcpProjectIdProvider();
+		String cloudPubSubPropertiesPrefix = GcpPubSubProperties.class.getAnnotation(ConfigurationProperties.class)
+				.value();
+		GcpPubSubProperties pubSubProperties = binder.bind(cloudPubSubPropertiesPrefix, GcpPubSubProperties.class)
+				.orElse(new GcpPubSubProperties());
+		Integer globalExecutorThreads = pubSubProperties.getSubscriber().getExecutorThreads();
+		Integer numThreads = globalExecutorThreads != null ? globalExecutorThreads
+				: PubSubConfiguration.DEFAULT_EXECUTOR_THREADS;
+		this.globalScheduler = createAndRegisterSchedulerBean(numThreads, "global-gcp-pubsub-subscriber",
+				"globalThreadPoolScheduler", beanDefinitionRegistry);
```

Comment:
`globalThreadPoolScheduler` should mention `Pubsub`. Are we going to have a different one for publishing?